make migrate command fails

The `make migrate` command failed with below traceback on Pyhton3.7 environment.

```
python3 manage.py migrate --noinput
Traceback (most recent call last):
  File "manage.py", line 22, in <module>
    execute_from_command_line(sys.argv)
  File "/home/akgaikwad/projects/makegoalsdaily/venv/lib64/python3.7/site-packages/django/core/management/__init__.py", line 364, in execute_from_command_line
    utility.execute()
  File "/home/akgaikwad/projects/makegoalsdaily/venv/lib64/python3.7/site-packages/django/core/management/__init__.py", line 338, in execute
    django.setup()
  File "/home/akgaikwad/projects/makegoalsdaily/venv/lib64/python3.7/site-packages/django/__init__.py", line 27, in setup
    apps.populate(settings.INSTALLED_APPS)
  File "/home/akgaikwad/projects/makegoalsdaily/venv/lib64/python3.7/site-packages/django/apps/registry.py", line 85, in populate
    app_config = AppConfig.create(entry)
  File "/home/akgaikwad/projects/makegoalsdaily/venv/lib64/python3.7/site-packages/django/apps/config.py", line 94, in create
    module = import_module(entry)
  File "/usr/lib64/python3.7/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1006, in _gcd_import
  File "<frozen importlib._bootstrap>", line 983, in _find_and_load
  File "<frozen importlib._bootstrap>", line 967, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 677, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 728, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/akgaikwad/projects/makegoalsdaily/venv/lib64/python3.7/site-packages/django/contrib/admin/__init__.py", line 4, in <module>
    from django.contrib.admin.filters import (
  File "/home/akgaikwad/projects/makegoalsdaily/venv/lib64/python3.7/site-packages/django/contrib/admin/filters.py", line 10, in <module>
    from django.contrib.admin.options import IncorrectLookupParameters
  File "/home/akgaikwad/projects/makegoalsdaily/venv/lib64/python3.7/site-packages/django/contrib/admin/options.py", line 13, in <module>
    from django.contrib.admin import helpers, widgets
  File "/home/akgaikwad/projects/makegoalsdaily/venv/lib64/python3.7/site-packages/django/contrib/admin/widgets.py", line 152
    '%s=%s' % (k, v) for k, v in params.items(),
    ^
SyntaxError: Generator expression must be parenthesized
make: *** [Makefile:31: migrate] Error 1

```

--- a/Makefile
+++ b/Makefile
@@ -18,6 +18,10 @@ env-info:
 	uname -a
 	pip list
 
+.PHONY: initlogin
+initlogin:
+	python3 manage.py initlogin
+
 .PHONY: lint
 lint:
 	flake8 --ignore=E501,F401,F403,F405,F999 makegoalsdaily logtoday

--- a/README.md
+++ b/README.md
@@ -1,4 +1,4 @@
-# log-today
+# make-goals-daily
 
 #### Make contribution to your short term goals daily :)
 ##### And stay on track, just in 3 steps..

--- a/logtoday/views.py
+++ b/logtoday/views.py
@@ -131,7 +131,13 @@ class ReportGoalsProgress(TemplateView):
     def _goal_progress_percentage(self, goal_start_date, goal_end_date, t_weights, s_weights=None):
         if not s_weights:
             s_weights = 2
-        return (t_weights / ((goal_end_date - goal_start_date).days * s_weights)) * 100
+        percentage = 0.0
+        try:
+            percentage = (t_weights / ((goal_end_date - goal_start_date).days * s_weights)) * 100
+        except ZeroDivisionError:
+            # pass for now
+            pass
+        return percentage
 
     def get_context_data(self, **kwargs):
         context_data = super(ReportGoalsProgress, self).get_context_data(**kwargs)

--- a/requirements.txt
+++ b/requirements.txt
@@ -1,8 +1,8 @@
-appdirs==1.4.3
-Django==1.11.15
-django-debug-toolbar==1.8
-packaging==16.8
-pyparsing==2.2.0
-pytz==2017.2
-six==1.10.0
-sqlparse==0.2.3
+appdirs
+Django
+django-debug-toolbar
+packaging
+pyparsing
+pytz
+six
+sqlparse

